InternalDomain: Make sure that the generate is not too large

goldhoorn opened the pull request:
Java has a limitation of 512kb per bytecode-function.
The genrated code was too lager in my case.
This patch limits the createState function to 500 lines.
This limitation should then fits.
Additional functions are created and called automatically.

goldhoorn commented:
anyone here alive on this project?

svenschneider commented:
Hi Matthias,

thanks for the pull request. It was nice, if you could format the changes according to the surrounding code and update the pull request. I will put some comments in the commit. If it's fine with you, I can also integrate the changes?

Additionally, please note, that this a fork of the original JSHOP2 [1][2]. Therefore, you may also consider sending your update to original authors.

Best regards
Sven

[1] http://sourceforge.net/projects/shop/files/JSHOP2/
[2] https://www.cs.umd.edu/projects/shop/description.html

svenschneider commented on src/JSHOP2/InternalDomain.java:
```diff
@@ -585,6 +585,9 @@ public String generateProblemCode(LinkedList<Vector<Predicate>> states, LinkedLi
 
     //-- For each problem,
+    int counter = 0;
+    int helper_idx = 0;
```
The convention in the remaining code is camel-case variable naming: helperIdx

svenschneider commented on src/JSHOP2/InternalDomain.java:
```diff
@@ -585,6 +585,9 @@ public String generateProblemCode(LinkedList<Vector<Predicate>> states, LinkedLi
     int problemIdx = 0;
 
     //-- For each problem,
+    int counter = 0;
```
Maybe lineCounter is a more telling name?

svenschneider commented on src/JSHOP2/InternalDomain.java:
```diff
@@ -594,6 +597,13 @@ public String generateProblemCode(LinkedList<Vector<Predicate>> states, LinkedLi
       //-- For each predicate, in the initial world state of the problem
       for (Predicate p : state)
       {
+        if(counter++ == 500){
```
The surrounding code uses spaces in the following situations. Thus, I suggest to stay consistent here and in the following lines:
1. After keywords
2. Before opening braces
3. Around binary operators
4. Around assignment equals symbols

Additionally, a line of comment for this block may be helpful. The content can e.g. be similar to your message in the pull request.

goldhoorn commented:
I do not working longer on this. Feel free to take over (or not) the responsibility for this PR.

goldhoorn commented:
i found a minute, updated

goldhoorn committed:
InternalDomain: Make sure that the generate is not too large

Java has a limitation of 512kb per bytecode-function.
The genrated code was too lager in my case.
This patch limits the createState function to 500 lines.
This limitation should then fits.
Additional functions are created and called automatically.

svenschneider commented:
Hi Matthias,

thanks for the update. There were still some minor formatting issues and a short explanation missing (in my opinion). Thus, I updated your commit accordingly, see commit b93c76ccb0e211f185dd2f4c446beb986acaade4. Are you fine if I merge that commit?

Best regards
Sven

goldhoorn commented:
Hi Sven,

Sure feel free to push this and close the PR afterwards.
I send my patch to the initial dev's don't know if this will go in the 
next release.

On 17.08.2015 10:37, Sven Schneider wrote:

> Hi Matthias,
> 
> thanks for the update. There were still some minor formatting issues 
> and a short explanation missing (in my opinion). Thus, I updated your 
> commit accordingly, see commit b93c76c 
> https://github.com/mas-group/jshop2/commit/b93c76ccb0e211f185dd2f4c446beb986acaade4. 
> Are you fine if I merge that commit?
> 
> Best regards
> Sven
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/mas-group/jshop2/pull/1#issuecomment-131730762.

svenschneider commented:
Hi Matthias,

great and thanks for the support! The changes have been integrated now (see #2 ). For the original version it's up to the authors to decide if they want to integrate the patch.

Best regards
Sven
